helpick-service/src/main/java/com/infoshare/service/NeedRequestService.java

```diff
@@ -29,42 +24,64 @@ public NeedRequestService(DB db) {
         this.db = db;
     }
 
-    public void createNeedRequest(String name, String location, String phone, TypeOfHelp typeOfHelp) {
+    public void createNeedRequest(String name, String location, String phone, TypeOfHelp typeOfHelp, UUID uuid) {
         PersonInNeed personInNeed = new PersonInNeed(name, location, phone);
-        NeedRequest needRequest = NeedRequest
-                .create(typeOfHelp, personInNeed);
-        db.saveNeedRequest(needRequest);
+        NeedRequest needRequest;
+        if (null == uuid) {
+            needRequest = NeedRequest
+                    .create(typeOfHelp, personInNeed);
+            db.saveNeedRequest(needRequest);
+        } else {
+            Optional<NeedRequest> needRequestOptional = db.getNeedRequests().stream()
+                    .filter(needRequest1 -> needRequest1.getUuid().equals(uuid))
+                    .findAny();
+            if (needRequestOptional.isPresent()) {
+                needRequest = needRequestOptional.get();
+                needRequest.setPersonInNeed(personInNeed);
+                needRequest.setTypeOfHelp(typeOfHelp);
+                db.saveNeedRequest(needRequest);
+            }
+        }
+
+    }
+
+    public void createNeedRequest(String name, String location, String phone, TypeOfHelp typeOfHelp) {
+        createNeedRequest(name, location, phone, typeOfHelp, null);
```

Comment:
jako pytanie: dlaczego przakazujemy `null`? :) czy może lepiej Optional'a?

Reply:
Zmienilam kod na dwie oddzielne metody za sugestia z https://www.freecodecamp.org/news/optional-in-java-and-anti-patterns-using-it-7d87038362ba/